### ISSUE TITLE
Can't showShareSheet from UIBarButtonItem in iPad

### ISSUE THREAD

**Issue by gongqinwei** [BranchMetrics/iOS-Deferred-Deep-Linking-SDK#286]
The BranchUniversalObject method `- (void)showShareSheetWithLinkProperties:(BranchLinkProperties *)linkProperties andShareText:(NSString *)shareText fromViewController:(UIViewController *)viewController andCallback:(callback)callback;` only presents popover from a `sourceView`. 

This is fine with iPhones. But for iPads, this would blindly present the popover from the top-left corner of the presenting view, regardless where the "anchor" (e.g. an "Action" UIBarButtonItem) is placed on the screen, i.e. users click the bar button on the top-right corner, but the popover shows up from the top-left corner.

It'd be great if the following method could be added to the SDK:
```
- (void)showShareSheetWithLinkProperties:(BranchLinkProperties *)linkProperties andShareText:(NSString *)shareText fromViewController:(UIViewController *)viewController anchor:(UIBarButtonItem *)anchor andCallback:(callback)callback {
    UIActivityItemProvider *itemProvider = [self getBranchActivityItemWithLinkProperties:linkProperties];
    NSMutableArray *items = [NSMutableArray arrayWithObject:itemProvider];
    if (shareText) {
        [items addObject:shareText];
    }
    UIActivityViewController *shareViewController = [[UIActivityViewController alloc] initWithActivityItems:items applicationActivities:nil];
    
    UIViewController *presentingViewController;
    if (viewController && [viewController respondsToSelector:@selector(presentViewController:animated:completion:)]) {
        presentingViewController = viewController;
    }
    else if ([[[[UIApplication sharedApplication].delegate window] rootViewController] respondsToSelector:@selector(presentViewController:animated:completion:)]) {
        presentingViewController = [[[UIApplication sharedApplication].delegate window] rootViewController];
    }
    
    if (linkProperties.controlParams[BRANCH_LINK_DATA_KEY_EMAIL_SUBJECT]) {
        @try {
            [shareViewController setValue:linkProperties.controlParams[BRANCH_LINK_DATA_KEY_EMAIL_SUBJECT] forKey:@"subject"];
        }
        @catch (NSException *exception) {
            NSLog(@"[Branch warning] Unable to setValue 'emailSubject' forKey 'subject' on UIActivityViewController.");
        }
    }
    
    if (anchor) {
        if ([shareViewController respondsToSelector:@selector(popoverPresentationController)]) {
            shareViewController.popoverPresentationController.barButtonItem = anchor;
        }
        [presentingViewController presentViewController:shareViewController animated:YES completion:callback];
    }
    else {
        NSLog(@"[Branch warning, fatal] No view controller is present to show the share sheet. Aborting.");
    }
}
```

Thanks!
P.S. tested

**Comment by ahmednawar** [comment]
Thanks @gongqinwei for reporting this and providing a solution! We'll test this and include it in the upcoming releases.

**Comment by ahmednawar** [comment]
@gongqinwei we fixed this problem on PR #296. Will ship it on the next release.
